### PR TITLE
changes to support neo4j 5.10.0

### PR DESCRIPTION
**Pull request by PranjaliParnerkar**
@paltusplintus Please review!
@KWLgsk @Vidhisri56 FYI.

This PR is for this [PBI](https://dev.azure.com/DevOps-RD/Clinical%20Linked%20Data%20Experiments/_sprints/taskboard/Clinical%20Linked%20Data%20Experiments%20Team/Clinical%20Linked%20Data%20Experiments/Sprint%2011.7).

The changes have been made as per neo4j 5.10.0 with backward compatibility.

NOTE: The test cases for rdf_support are not working currently as it requires n10 plugin support which is not there currently so unable to test it.

### CODE REVIEW

File: tests/test_json.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def test_export_dbase_json(db):
     assert result['properties'] == 3
     expected_json = f'[{{"type":"node","id":"{node_id1}","labels":["User"],"properties":{{"name":"Eve"}}}},\n' \
         f' {{"type":"node","id":"{node_id2}","labels":["User"],"properties":{{"name":"Adam","age":30}}}},\n' \
-        f' {{"id":"{rel_id_1}","type":"relationship","label":"LOVES","start":{{"id":"{node_id1}","labels":["User"]}},"end":{{"id":"{node_id2}","labels":["User"]}}}}\n]'
+        f' {{"type":"relationship","id":"{rel_id_1}","label":"LOVES","start":{{"id":"{node_id1}","labels":["User"],"properties":{{"name":"Eve"}}}},"end":{{"id":"{node_id2}","labels":["User"],"properties":{{"name":"Adam","age":30}}}}}}\n]'
```
Comment (paltusplintus) [inline tests/test_json.py:63]: make sure the test also works on 4.x

**Review by paltusplintus — CHANGES_REQUESTED** [review]
@PranjaliParnerkar Thanks for the updates! please see some proposals below.

File: neointerface/neointerface.py [hunk 2]
```diff
@@ -787,7 +795,14 @@ def get_constraints(self) -> pd.DataFrame:
 
         :return:  A (possibly-empty) Pandas dataframe
         """
-        q = """
+        if neo4j_driver_version=="5.10.0":
+            q="""
+           SHOW CONSTRAINTS
+           yield name, type
```
[the remark this answers is not in the document]
Comment (PranjaliParnerkar) [reply neointerface/neointerface.py:803]: │"id"│"name"│"type"│"entityType"│"labelsOrTypes"│"properties"│"ownedIndex"│"propertyType" |

These are the details we are getting from SHOW CONSTRAINTS. Shall I use all of them?

**Comment by PranjaliParnerkar** [comment]
@paltusplintus I have made the changes as per comments.

**Comment by PranjaliParnerkar** [comment]
There will be few more changes in Neointerface, got some issue while testing tab2neo. Moving this to draft for now.

**Comment by PranjaliParnerkar** [comment]
@paltusplintus Please review!

**Review by paltusplintus — APPROVED** [review]

**Pull request merged**